controllers/withwatcher/kyma_controller_test.go

```diff
@@ -21,90 +27,212 @@ import (
+		Eventually(IsSKRWebhookConditionReasonReady(suiteCtx, controlPlaneClient, kymaObjKey),
 			Timeout, Interval).Should(BeTrue())
+		Expect(deploy.CheckWebhookCABundleConsistency(suiteCtx, runtimeClient, kymaObjKey)).To(Succeed())
 	})
 
-	It("kyma reconciler installs watcher helm chart with correct webhook config when watcher specs are updated",
-		func() {
-			Skip("failing because of manifest caching invalidation")
-			latestWatcher := &v1alpha1.Watcher{}
-			Expect(controlPlaneClient.Get(suiteCtx, client.ObjectKeyFromObject(watcherCrForKyma), latestWatcher)).
-				To(Succeed())
-			latestWatcher.Spec.LabelsToWatch["new-key"] = "new-value"
-			Expect(controlPlaneClient.Update(suiteCtx, latestWatcher)).To(Succeed())
-			latestKyma := &v1alpha1.Kyma{}
-			Expect(controlPlaneClient.Get(suiteCtx, client.ObjectKeyFromObject(kyma), latestKyma)).To(Succeed())
-			latestKyma.Spec.Channel = v1alpha1.ChannelFast
-			Expect(controlPlaneClient.Update(suiteCtx, latestKyma)).To(Succeed())
-			webhookConfig := &admissionv1.ValidatingWebhookConfiguration{}
-			Eventually(isWebhookDeployed(suiteCtx, runtimeClient, webhookConfig), Timeout, Interval).Should(Succeed())
-			Eventually(IsKymaInState(suiteCtx, controlPlaneClient, kyma.GetName(), v1alpha1.StateReady),
-				Timeout, Interval).Should(BeTrue())
-			Expect(isWebhookConfigured(latestWatcher, webhookConfig)).To(BeTrue())
-		})
+	It("Watcher helm chart caching works as expected", func() {
+		labelKey := "new-key"
+		labelValue := "new-value"
+		watcherCrForKyma.Spec.LabelsToWatch[labelKey] = labelValue
+		Expect(controlPlaneClient.Update(suiteCtx, watcherCrForKyma)).To(Succeed())
+		By("waiting for watcher CR labelsToWatch to be updated")
+		Eventually(isWatcherCrLabelUpdated(client.ObjectKeyFromObject(watcherCrForKyma),
+			labelKey, labelValue), Timeout, Interval).Should(BeTrue())
+		By("updating kyma channel to trigger its reconciliation")
+		kyma.Spec.Channel = v1alpha1.ChannelFast
+		Expect(controlPlaneClient.Update(suiteCtx, kyma)).To(Succeed())
+		By("waiting for the kyma update event to be processed")
+		time.Sleep(2 * time.Second)
+		By("waiting for skr webhook condition to be ready")
+		Eventually(IsSKRWebhookConditionReasonReady(suiteCtx, controlPlaneClient, kymaObjKey),
+			Timeout, Interval).Should(BeTrue())
+		Expect(runtimeClient.Get(suiteCtx, client.ObjectKey{
+			Namespace: metav1.NamespaceDefault,
+			Name:      deploy.ResolveSKRChartResourceName(deploy.WebhookConfigNameTpl, kymaObjKey),
+		}, webhookConfig)).To(Succeed())
+		Expect(isWebhookConfigured(watcherCrForKyma, webhookConfig, kyma.Name)).To(Succeed())
+
+		caBundleValueBeforeUpdate := webhookConfig.Webhooks[0].ClientConfig.CABundle
+		By("updating kyma channel to trigger its reconciliation")
+		kyma.Spec.Channel = v1alpha1.ChannelRapid
+		Expect(controlPlaneClient.Update(suiteCtx, kyma)).To(Succeed())
+		By("waiting for the kyma update event to be processed")
+		time.Sleep(2 * time.Second)
```

Comment:
could we remove `time.Sleep`?

Reply:
I am not sure we can. I tried different post checks without success. what do you suggest?